src/nifreeze/model/dmri.py

```diff
@@ -210,36 +207,26 @@ def __init__(
         self._th_high = th_high
         self._detrend = detrend
 
-    def fit_predict(self, index, *_, **kwargs):
+    def fit_predict(self, index: int, *_, **kwargs):
         """Return the average map."""
 
-        bvalues = self._dataset.gradients[:, -1]
-        bcenter = bvalues[index]
-
-        shellmask = np.ones(len(self._dataset), dtype=bool)
-
-        # Keep only bvalues within the range defined by th_high and th_low
-        shellmask[index] = False
```

Comment:
@oesteban While writing the test, I have realized that this statement should be removed: we want to keep the value at the `index` position, right?

Reply:
I removed the statement. Consequently, the exception here is not not raised:
https://github.com/nipreps/nifreeze/pull/71/files#diff-2197e776366f74ac177f201dce923642554f122630fabab0b45063f2f6cf1832R230-R231

https://github.com/nipreps/nifreeze/actions/runs/14149369168/job/39640631654?pr=71#step:11:539

So I am wondering about the rationale behind assigning `False` to the index.

Reply:
As long as `dwi_select_shells` does this inside (which I imagine it does so), yes, you need to remove this because it's moved into the new function.

Reply:
It does not, please have a look at the definition https://github.com/nipreps/nifreeze/pull/71/files#diff-2197e776366f74ac177f201dce923642554f122630fabab0b45063f2f6cf1832R183

Reply:
> So I am wondering about the rationale behind assigning `False` to the index.

Okay, I just came back to this and now I understand. The idea for this method is: we want to leave one volume out at each iteration, so you want to mark that specific volume for dropping in the returned mask. Does that make sense?

Reply:
The latest forced push d30f522 solves the questions here. Resolving.